rhel/SPECS/dbaas-controller.spec

```diff
@@ -0,0 +1,64 @@
+%undefine _missing_build_ids_terminate_build
+%global _dwz_low_mem_die_limit 0
+
+# do not strip debug symbols
+%global debug_package   %{nil}
+
+%global provider	github
+%global provider_tld	com
+%global project		percona-platform
+%global repo		dbaas-controller
+%global provider_prefix	%{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path	%{provider_prefix}
+%global commit		41d72afe43a88f522bdd35a4281a447424f18055
+%global shortcommit	%(c=%{commit}; echo ${c:0:7})
+%define build_timestamp %(date -u +"%y%m%d%H%M")
+%define release         1
+%define rpm_release     %{release}.%{build_timestamp}.%{shortcommit}%{?dist}
+
+%global install_golang 0
+
+Name:		%{repo}
+Version:	%{version}
+Release:	%{rpm_release}
+Summary:	Simplified API for managing Percona Kubernetes Operators
+
+License:	AGPLv3
+URL:		https://%{provider_prefix}
+Source0:	https://%{provider_prefix}/archive/%{commit}/%{repo}-%{shortcommit}.tar.gz
+
+%if %{install_golang}
+BuildRequires:   golang >= 1.14.0
+%endif
+
+%description
+dbaas-controller exposes a simplified API for managing Percona Kubernetes Operators
+See the PMM docs for more information.
+
+
+%prep
+%setup -q -n %{repo}-%{commit}
+mkdir -p src/%{provider}.%{provider_tld}/%{project}
+ln -s $(pwd) src/%{provider_prefix}
```

Comment:
https://jira.percona.com/browse/PMM-5452

In that particular case we don't have to build this code in GOPATH, so we can build in /tmp/foo if we want to. But no symlinks should be used. See here: https://golang.org/doc/code.html

(other PMM components still use GOPATH: https://golang.org/doc/gopath_code.html)

Reply:
@AlekSi the problem is, that `%setup` step creates build directory and extracts tarball with sources into it. You can't override build directory. As I said couple of month ago, we can move compilation steps outside of RPM spec into pipeline script, build applications and copy resulting binaries using spec.

Reply:
So what? dbaas-controller can be build in any directory. We should not use `ln`. Please re-read what I wrote.

Reply:
I understand, that controller can be built in any directory. The problem is, that rpmbuild forces build directory and symlinks are used to make **go build** work properly with these limitations.